java/test/src/main/java/org/ray/api/test/ReferenceCountingTest.java

```diff
@@ -136,4 +138,20 @@ public void testDependencyRefCounts() {
     ((RayObjectImpl<?>) result).removeLocalReference();
     checkRefCounts(ImmutableMap.of());
   }
+
+  public void testBasicNestedIds() {
+    RayObject<byte[]> inner = Ray.put(new byte[40 * 1024 * 1024]);
+    RayObject<List<RayObject<byte[]>>> outer = Ray.put(Collections.singletonList(inner));
+
+    byte[] innerObjecrIdBytes = inner.getId().getBytes();
+    inner = null;
+    System.gc();
+
+    for (int i = 0; i < 100; i++) {
+      Ray.put(new byte[40 * 1024 * 1024]);
+    }
+
+    inner = new RayObjectImpl<>(new ObjectId(innerObjecrIdBytes));
+    Assert.assertNotNull(inner.get());
```

Comment:
Could you explain why the first object is still available? Is it because the outer object is guaranteed to not be garbage collected? I don't see any references to the outer object at this line. Is it possible that the outer object is also garbage collected when you call `System.gc()`?

It would be better you can write some comments in the code.